src/cpp/rtps/builtin/discovery/participant/PDPListener.cpp

```diff
@@ -147,13 +152,23 @@ void PDPListener::onNewCacheChangeAdded(
                     RTPSParticipantListener* listener = parent_pdp_->getRTPSParticipant()->getListener();
                     if (listener != nullptr)
                     {
-                        std::lock_guard<std::mutex> cb_lock(parent_pdp_->callback_mtx_);
-                        ParticipantDiscoveryInfo info(*pdata);
-                        info.status = status;
+                        bool should_be_ignored = false;
+                        {
+                            std::lock_guard<std::mutex> cb_lock(parent_pdp_->callback_mtx_);
+                            ParticipantDiscoveryInfo info(*pdata);
+                            info.status = status;
+
+
+                            listener->onParticipantDiscovery(
```

Comment:
Should we be doing something similar on [PDPServerListener](https://github.com/eProsima/Fast-DDS/blob/c8a3bee222957f7f28fd488d790636567f5987df/src/cpp/rtps/builtin/discovery/participant/PDPServerListener.cpp#L370)?

Reply:
Done